### PR TITLE
block_during_io: Fetch the drive letters again after reboot

### PR DESCRIPTION
**Pull request by YongxueHong**
The data disk letters will be changed after reboot in windows, 
so fetch the letters again.

ID： 1731895
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

File: qemu/tests/block_during_io.py [hunk 1]
```diff
@@ -140,5 +141,14 @@ def run_bg_test(target, args=(), kwargs={}):
     bg_test_thread.join(bg_test_thread_timeout)
 
     if not shutdown_vm:
+        if with_data_disks and windows:
+            # XXX: The data disk letters will be changed after reboot in windows.
```
Comment (XueqiangWei) [inline qemu/tests/block_during_io.py]: Just \"system_reset\", hit this issue, please change \"reboot\" to \"system_reset\".

**Comment by kimi1978wy** [comment]
Error occurred when running on win10-32 guest:

 (01/50) repeat1.Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset: ERROR: Shell process terminated while waiting for command to complete: 'fsutil fsinfo drives'    (status: 1,    output: 'Ncat: Connection reset by peer.\n')

**Comment by YongxueHong** [comment]
> Error occurred when running on win10-32 guest:
> 
> (01/50) repeat1.Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset: ERROR: Shell process terminated while waiting for command to complete: 'fsutil fsinfo drives' (status: 1, output: 'Ncat: Connection reset by peer.\n')

Sorry, I will fix it right now.

**Comment by YongxueHong** [comment]
@hereischen  pls help me to review, thanks.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by kimi1978wy** [comment]
@YongxueHong  I still failed with this patch
(02/31) repeat1.Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_qmp.with_system_reset: ERROR: The iozone stress is not alive. (217.96 s)

**Comment by peixiu** [comment]
Tested with scsi, it can be passed after run 60 times, pasted part of result:

 (01/60) repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_qmp.with_system_reset.q35: PASS (435.51 s)
 (02/60) repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_reboot.q35: PASS (378.00 s)
 (03/60) repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_shutdown.q35: PASS (123.94 s)
 (04/60) repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset.q35: PASS (489.63 s)
 (05/60) repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_reboot.q35: PASS (451.69 s)
 (06/60) repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_shutdown.q35: PASS (123.32 s)

**Comment by XueqiangWei** [comment]
Tested with guest rhel8 and win2019, all passed. part of results as below:

CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
'01-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_qmp.with_system_reset.q35' Finshed 2019-07-31 08:23:15 2019-07-31 08:33:55 PASS 639.672371
'02-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_reboot.q35' Finshed 2019-07-31 08:33:55 2019-07-31 08:44:00 PASS 604.68263
'03-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_shutdown.q35' Finshed 2019-07-31 08:44:00 2019-07-31 08:45:54 PASS 114.520524
'04-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset.q35' Finshed 2019-07-31 08:45:54 2019-07-31 08:56:19 PASS 624.216656
'05-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_reboot.q35' Finshed 2019-07-31 08:56:19 2019-07-31 09:06:06 PASS 587.297974
'06-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_shutdown.q35' Finshed 2019-07-31 09:06:06 2019-07-31 09:08:11 PASS 125.314613

'11-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_qmp.with_system_reset.q35' Finshed 2019-07-31 09:29:30 2019-07-31 09:43:15 PASS 825.316878
'12-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_reboot.q35' Finshed 2019-07-31 09:43:15 2019-07-31 09:51:12 PASS 477.332101
'13-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_shutdown.q35' Finshed 2019-07-31 09:51:13 2019-07-31 09:53:55 PASS 162.868127
'14-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset.q35' Finshed 2019-07-31 09:53:55 2019-07-31 10:03:12 PASS 556.992074
'15-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_reboot.q35' Finshed 2019-07-31 10:03:13 2019-07-31 10:11:45 PASS 512.122656
'16-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_shutdown.q35' Finshed 2019-07-31 10:11:45 2019-07-31 10:14:55 PASS 190.582798

**Comment by kimi1978wy** [comment]
Tried on win7-64/win8-32/Win10-32, it only shows "IOZONE.EXE" when checking the process.
Tried on Win2019, it shows both "iozone.exe" and  "IOZONE.EXE".
It running successfully and no error on each guest os. So I think it can check "IOZONE.EXE" instead. What do you think? @YongxueHong

**Comment by YongxueHong** [comment]
> Tried on win7-64/win8-32/Win10-32, it only shows "IOZONE.EXE" when checking the process.
> Tried on Win2019, it shows both "iozone.exe" and "IOZONE.EXE".
> It running successfully and no error on each guest os. So I think it can check "IOZONE.EXE" instead. What do you think? @YongxueHong

How about you? @XueqiangWei  @peixiu

**Comment by peixiu** [comment]
@YongxueHong it's ok for me~

**Comment by peixiu** [comment]
@YongxueHong and no diff the Upper and Lower should match all situations~

**Comment by XueqiangWei** [comment]
@YongxueHong it's ok for me.

**Comment by kimi1978wy** [comment]
Tested with virtio-blk, it can be passed with only one "iozone.exe" process (Win8-32/win10-32)
 (1/6) Host_RHEL.m7.u6.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_qmp.with_system_reset: PASS (953.32 s)
 (2/6) Host_RHEL.m7.u6.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_reboot: PASS (732.51 s)
 (3/6) Host_RHEL.m7.u6.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_shutdown: PASS (116.63 s)

**Comment by peixiu** [comment]
Tested on win7-64 guest, 3 cases with system_disk can be passed, but 3 case for data_disks all failed:

 (01/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_qmp.with_system_reset.q35: PASS (371.65 s)
 (02/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_reboot.q35: PASS (308.93 s)
 (03/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_shutdown.q35: PASS (81.47 s)
 (04/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset.q35: ERROR: No found the corresponding drive letters in 1 disks. (176.35 s)
 (05/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_reboot.q35: ERROR: Could not get exit status of command: 'echo list disk > disk_4S5l && echo select disk 1 >> disk_4S5l &&  echo create partition primary size=40955 >> disk_4S5l &&  echo exit >> disk_4S5l && diskpart /s disk_4S5l && del /f disk_4S5l'    (output: '\nMicrosof... (219.15 s)
 (06/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_shutdown.q35: ERROR: Could not get exit status of command: 'echo list disk > disk_yqfr && echo select disk 1 >> disk_yqfr &&  echo create partition primary size=40955 >> disk_yqfr &&  echo exit >> disk_yqfr && diskpart /s disk_yqfr && del /f disk_yqfr'    (output: '\nMicrosof... (225.60 s)

**Comment by YongxueHong** [comment]
> Tested on win7-64 guest, 3 cases with system_disk can be passed, but 3 case for data_disks all failed:
> 
> (01/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_qmp.with_system_reset.q35: PASS (371.65 s)
> (02/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_reboot.q35: PASS (308.93 s)
> (03/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_shutdown.q35: PASS (81.47 s)
> (04/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset.q35: ERROR: No found the corresponding drive letters in 1 disks. (176.35 s)
> (05/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_reboot.q35: ERROR: Could not get exit status of command: 'echo list disk > disk_4S5l && echo select disk 1 >> disk_4S5l && echo create partition primary size=40955 >> disk_4S5l && echo exit >> disk_4S5l && diskpart /s disk_4S5l && del /f disk_4S5l' (output: '\nMicrosof... (219.15 s)
> (06/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_shutdown.q35: ERROR: Could not get exit status of command: 'echo list disk > disk_yqfr && echo select disk 1 >> disk_yqfr && echo create partition primary size=40955 >> disk_yqfr && echo exit >> disk_yqfr && diskpart /s disk_yqfr && del /f disk_yqfr' (output: '\nMicrosof... (225.60 s)

For 4th case is a product bug, for 5th~6th cases are another product bug, wyu should know about them.

**Comment by kimi1978wy** [comment]
> > (04/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset.q35: ERROR: No found the corresponding drive letters in 1 disks. (176.35 s)
> > (05/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_reboot.q35: ERROR: Could not get exit status of command: 'echo list disk > disk_4S5l && echo select disk 1 >> disk_4S5l && echo create partition primary size=40955 >> disk_4S5l && echo exit >> disk_4S5l && diskpart /s disk_4S5l && del /f disk_4S5l' (output: '\nMicrosof... (219.15 s)
> > (06/14) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_shutdown.q35: ERROR: Could not get exit status of command: 'echo list disk > disk_yqfr && echo select disk 1 >> disk_yqfr && echo create partition primary size=40955 >> disk_yqfr && echo exit >> disk_yqfr && diskpart /s disk_yqfr && del /f disk_yqfr' (output: '\nMicrosof... (225.60 s)
> 
> For 4th case is a product bug, for 5th~6th cases are another product bug, wyu should know about them.

They are the same product bug. It is because, we cannot format data disk after "system_reset while running iozone" on win7 and win2008R2 , but the first time before system_reset, it can format successfully, so it occurs error " No found the corresponding drive letters in 1 disks"

**Comment by kimi1978wy** [comment]
LGTM
Acked-by: Yu Wang <wyu@redhat.com>

**Comment by peixiu** [comment]
Tested with win2016 again, all passed:

 (1/6) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_qmp.with_system_reset.q35: PASS (448.49 s)
 (2/6) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_reboot.q35: PASS (356.32 s)
 (3/6) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.system_disk.send_shell.with_shutdown.q35: PASS (122.10 s)
 (4/6) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_qmp.with_system_reset.q35: PASS (444.60 s)
 (5/6) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_reboot.q35: PASS (396.98 s)
 (6/6) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_during_io.iozone_stress.data_disks.send_shell.with_shutdown.q35: PASS (120.00 s)

Acked-by: Peixiu Hou phou@redhat.com

**Comment by XueqiangWei** [comment]
Acked-by xuwei@redhat.com

**Pull request merged**